src/ralph/admin/autocomplete.py

```diff
@@ -70,12 +73,21 @@ def dispatch(self, request, *args, **kwargs):
 
             def get_queryset(self, user):
                 queryset = self.model._default_manager.all()
-                if self.query:
-                    qs = [
-                        Q(**{'{}__icontains'.format(field): self.query})
-                        for field in search_fields
-                    ]
-                    queryset = queryset.filter(reduce(operator.or_, qs))
+                if issubclass(self.model, Polymorphic):
+                    bs_ids = []
+                    for related_model in self.model.get_related_models():
+                        bs_ids.extend(self.model.get_base_ids(
+                            related_model,
+                            self.query
+                        ))
+                    queryset = queryset.filter(pk__in=bs_ids)
```

Comment:
this conflicts a little bit with permissions per object.The simplest way IMO will be to add some logic to `get_base_ids` method (since i suggested to move it here, it will not conflict anyway with Polymorphic), pass there user and check there if related model is subclass of PermissionsForObjectMixin - if yes, then `_get_objects_for_user` should be called